### ISSUE TITLE
Bootsnap not compiling on branch main ( FreeBSD 10.0 ) [ fatal error: sys/xattr.h: No such file or directory #include <sys/xattr.h> ]

### ISSUE THREAD

**Issue by HugeTerrificSpider** [Shopify/bootsnap#32]
branch: master
OS: 10.3-STABLE FreeBSD

since bootsnap was added to discourse on May 11 when i try to

    env GYP_DEFINES="make_clang_dir=" CXX=/usr/bin/clang++ CC=/usr/bin/clang bundle install --without development test

I get error

    (...)
    Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /usr/home/spider/.rvm/gems/ruby-2.3.3/gems/bootsnap-0.2.14/ext/bootsnap
    /home/spider/.rvm/rubies/ruby-2.3.3/bin/ruby -r ./siteconf20170515-61851-u8spxp.rb extconf.rb
    creating Makefile

    current directory: /usr/home/spider/.rvm/gems/ruby-2.3.3/gems/bootsnap-0.2.14/ext/bootsnap
    make "DESTDIR=" clean

    current directory: /usr/home/spider/.rvm/gems/ruby-2.3.3/gems/bootsnap-0.2.14/ext/bootsnap
    make "DESTDIR="
    compiling bootsnap.c
    bootsnap.c:3:23: fatal error: sys/xattr.h: No such file or directory
     #include <sys/xattr.h>
                           ^
    compilation terminated.
    *** Error code 1

    Stop.
    make: stopped in /usr/home/spider/.rvm/gems/ruby-2.3.3/gems/bootsnap-0.2.14/ext/bootsnap

    make failed, exit code 1

    Gem files will remain installed in /home/spider/.rvm/gems/ruby-2.3.3/gems/bootsnap-0.2.14 for inspection.
    Results logged to /home/spider/.rvm/gems/ruby-2.3.3/extensions/x86_64-freebsd-10/2.3.0/bootsnap-0.2.14/gem_make.out

    An error occurred while installing bootsnap (0.2.14), and Bundler cannot continue.
    Make sure that `gem install bootsnap -v '0.2.14'` succeeds before bundling.

**Comment by SamSaffron** [comment]
Maybe if xattr.h does not exist it should just skip compiling that file? it can test for it prior to building.

**Comment by madmax** [comment]
FREEBSD has sys/extattr.h

**Comment by burke** [comment]
It does, but it's a different API too, not just a different header file name. There's already some conditional junk around darwin/linux API variation, so it wouldn't be horrible to add a variant for FreeBSD but it's not a high priority for me personally. I'd accept a PR as long as it doesn't add more complexity than I'm currently imagining it should.

**Comment by SamSaffron** [comment]
@burke a first phase PR here would simply be to make this a no-op in freebsd, so, at least it compiles

**Comment by burke** [comment]
Should be fixed by https://github.com/Shopify/bootsnap/pull/43, if you want to test it out that would help get it shipped sooner.

**Comment by burke** [comment]
@HugeTerrificSpider curious to know if 0.3.0.pre works for you.

**Comment by HugeTerrificSpider** [comment]
>Should be fixed by #43, if you want to test it out that would help get it shipped sooner.
Ok, I'm downloading it now to test if I can compile it.

**Comment by HugeTerrificSpider** [comment]
```
$ gem build bootsnap.gemspec
$ gem install bootsnap-0.3.0.pre.gem 
Building native extensions.  This could take a while...
Successfully installed bootsnap-0.3.0.pre
Parsing documentation for bootsnap-0.3.0.pre
Installing ri documentation for bootsnap-0.3.0.pre
Done installing documentation for bootsnap after 0 seconds
1 gem installed
```
@burke 
Seems like installation works fine. Anything else to test?

**Comment by dgmora** [comment]
I was hitting #33  with docker and `phusion/passenger-ruby23` image but after trying with 0.3.0.pre I was able to give bootsnap a try. Great!

**Comment by burke** [comment]
Ok, I've released 0.3.0 with full FreeBSD and Linux support.
